Std/Data/List/Basic.lean

```diff
@@ -823,6 +823,40 @@ inductive Forall₂ (R : α → β → Prop) : List α → List β → Prop
 
 attribute [simp] Forall₂.nil
 
+/--
+Decision procedure for `Forall₂ R l₁ l₂`
+-/
+def Forall₂.dec (R : α → β → Prop) [∀ a b, Decidable (R a b)] : List α → List β → Bool
+  | [], [] => true
+  | a::as, b::bs =>
+    if R a b then
+      dec R as bs
+    else false
+  | _, _ => false
+
+theorem Forall₂.dec_iff {R : a → b → Prop} [∀ a b, Decidable (R a b)] :
+    ∀ l₁ l₂, Forall₂.dec R l₁ l₂ ↔ Forall₂ R l₁ l₂
+  | [], [] => by simp [dec]
+  | a::as, b::bs => by
+    if h : R a b then
+      simp [dec, h, dec_iff]
+      constructor <;> intro h'
+      exact Forall₂.cons h h'
+      have .cons _ tail := h'
+      exact tail
```

Comment:
I think it would make sense to upstream `forall₂_cons` from mathlib in this PR, which can be golfed to
```
theorem forall₂_cons {R : α → β → Prop} {a b as bs} :
    Forall₂ R (a :: as) (b :: bs) ↔ R a b ∧ Forall₂ R as bs :=
  ⟨fun | .cons h tail => ⟨h, tail⟩, fun ⟨head, tail⟩ => Forall₂.cons head tail⟩
```

Reply:
should it maybe be called `Forall₂.cons_iff` or your name? Like, should it be under the `Forall₂` namespace or not

Reply:
I guess if that's already used I'll go with that